superdesk/data_updates/00018_20190801-145234_archive.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; -*-
+# This file is part of Superdesk.
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+#
+# Author  : petr
+# Creation: 2019-08-01 14:52
+
+from superdesk import get_resource_service
+from superdesk.commands.data_updates import DataUpdate
+
+
+class DataUpdate(DataUpdate):
+
+    resource = 'archive'
+
+    def forwards(self, mongodb_collection, mongodb_database):
+        for resource in ('archive', 'published'):
+            service = get_resource_service(resource)
+            for item in mongodb_database[resource].find({'type': 'text', 'associations': {'$gt': {}}}):
+                update = False
+                associations = {}
+                for key, val in item['associations'].items():
+                    if val and val.get('type') == 'text' and len(val.keys()) > 2:
```

Comment:
Why are you checking only the item of type `text`? We can add `picture`, `audio` or `video` as well inside related items of type `related_content`.

Reply:
good point, I'm not sure to be honest :wink: will fix

Reply:
actually it's safer like that, because otherwise it would also affect featured media etc, while text items can be only used as related and they were causing the performance issue this is trying to solve

Reply:
Yep, I do agree that text can only be used as related but a picture, audio or video can also be used as related. So, why not check if the association key is a related key of type `realted_content` not `media` then. If that makes sense :)

Reply:
makes sense, will check that :+1: